ospf6d/ospf6_abr.c

```diff
@@ -1199,10 +1216,21 @@ void ospf6_abr_examin_summary(struct ospf6_lsa *lsa, struct ospf6_area *oa)
 		break;
 	}
 
+	/* If the old entry is not updated and old entry not found or old entry
+	 * does not match with the new entry then add the new route
+	 */
 	if (old_entry_updated == false) {
+		if (old == NULL)
+			add_route = true;
+		else if ((old->type != route->type)
+			 || (old->path.type != route->path.type))
+			add_route = true;
+	}
+
+	if (add_route == true) {
```

Comment:
don't compare to true, just use the boolean itself

Reply:
resolved